src/main/java/com/iridium/iridiumskyblock/utils/RandomAccessList.java

```diff
@@ -14,42 +16,96 @@
  */
 public class RandomAccessList<E> {
 
-    private final List<E> underlyingList = new ArrayList<>();
+    private final List<RandomAccessElement<E>> elements = new ArrayList<>();
```

Comment:
Why do we have a List of RandomAccessElements? I dont really like this, Feel like we should have a Hashmap instead?